File: kyulee/programmers-2/데이터분석.java
import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

public class 데이터분석 {
    public class Solution {
        public List<int[]> solution(int[][] data, String ext, int val_ext, String sort_by) {
            List<int[]> filteredData = new ArrayList<>();

            int extIndex = getIndex(ext);
            for (int[] entry : data) {
                if (entry[extIndex] < val_ext) {
                    filteredData.add(entry);
                }
            }

            int sortByIndex = getIndex(sort_by);
            filteredData.sort(Comparator.comparingInt(o -> o[sortByIndex]));

            return filteredData;
        }

        private int getIndex(String key) {
            switch (key) {
                case "code":
                    return 0;
                case "date":
                    return 1;
                case "maximum":
                    return 2;
                case "remain":
                    return 3;
                default:
                    throw new IllegalArgumentException("Invalid key: " + key);
            }
        }
    }
}

File: soyeong/programmers-2/데이터-분석.py
def solution(data, ext, val_ext, sort_by):
    answer = []
    # 기준 정보에 대한 인덱스 값을 dictionary로 정리
    dict = {"code":0, "date":1, "maximum":2, "remain":3}
    for d in data:
        # 기준 정보의 인덱스 값 할당
        value = d[dict[ext]]
        # 정보 기준값 이하
        if value <= val_ext:
            answer.append(d)
    # 정렬 조건을 대상으로 오름차순
    answer.sort(key = lambda x : x[dict[sort_by]])
    return answer